lib/inbox.rb

```diff
@@ -117,7 +117,7 @@ def url_for_authentication(redirect_uri, login_hint = '', options = {})
       if options[:trial] == true
         trialString = 'true'
       end
-      "https://#{@service_domain}/oauth/authorize?client_id=#{@app_id}&trial=#{trialString}&response_type=code&scope=email&login_hint=#{login_hint}&redirect_uri=#{redirect_uri}"
+      "https://#{@service_domain}/oauth/authorize?client_id=#{@app_id}&trial=#{trialString}&response_type=code&scope=email&login_hint=#{login_hint}&redirect_uri=#{redirect_uri}&state=#{options[:state]}"
```

Comment:
Great idea. How do you feel about me updating some of the other parameters (like login_hint) to also be omitted when they're blank?